src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionTests.ConnectionLifecycle.cs

```diff
@@ -380,6 +381,52 @@ public async Task SSEWaitsForResponseToStart()
                 }
             }
 
+            [Fact]
+            public async Task CanCancelStartingConnectionAfterNegotiate()
+            {
+                using (StartVerifiableLog())
+                {
+                    // Set up a SyncPoint within Negotiate, so we can verify
+                    // that the call has gotten that far
+                    var negotiateSyncPoint = new SyncPoint();
+                    var testHttpHandler = new TestHttpMessageHandler(autoNegotiate: false);
+                    testHttpHandler.OnNegotiate(async (request, cancellationToken) =>
+                    {
+                        // Wait here for the test code to cancel the "outer" token
+                        await negotiateSyncPoint.WaitToContinue();
+
+                        // Cancel
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        return ResponseUtils.CreateResponse(HttpStatusCode.OK);
+                    });
+
+                    await WithConnectionAsync(
+                        CreateConnection(testHttpHandler),
+                        async (connection) =>
+                        {
+                            // Kick off StartAsync, but don't wait for it
+                            var cts = new CancellationTokenSource();
+                            var startTask = connection.StartAsync(cts.Token);
+
+                            // Wait for the connection to get to the "WaitToContinue" call above,
+                            // which means it has gotten to Negotiate
+                            await negotiateSyncPoint.WaitForSyncPoint();
```

Comment:
We generally put `.OrTimeout();` on all `await` calls in tests to help detect test hangs.